src/Compiler/Service/BackgroundCompiler.fs

```diff
@@ -1372,14 +1372,7 @@ type internal BackgroundCompiler
             let diags =
                 loadClosure.LoadClosureRootFileDiagnostics
                 |> List.map (fun (exn, isError) ->
-                    FSharpDiagnostic.CreateFromException(
-                        exn,
-                        isError,
-                        range.Zero,
-                        false,
-                        options.OtherOptions |> Array.contains "--flaterrors",
-                        None
-                    ))
+                    FSharpDiagnostic.CreateFromException(exn, isError, false, options.OtherOptions |> Array.contains "--flaterrors", None))
```

Comment:
Could you please extract `options.OtherOptions |> Array.contains "--flaterrors"`, so it's not calculated many times?

Reply:
Done. Thanks!